src/netlink/nl_l3.cc

```
@@ -406,7 +406,18 @@ int nl_l3::del_l3_addr(struct rtnl_addr *a) {
 
   int rv = 0;
   int family = rtnl_addr_get_family(a);
-  struct rtnl_link *link = rtnl_addr_get_link(a);
+
+  // The link needs to be obtained from the cache, as the link reference
+  // may be outdated
+  struct rtnl_link *link = nl->get_link(rtnl_addr_get_ifindex(a), AF_UNSPEC);
+
+  if (link == nullptr) {
+    LOG(ERROR) << __FUNCTION__
+               << ": unknown link ifindex=" << rtnl_addr_get_ifindex(a)
+               << " of addr=" << OBJ_CAST(a);
+    return rv;
```

rv is zero at this place, you might want to return an error value here